doc/getting-started/apl-nuc.rst

```diff
@@ -178,8 +178,13 @@ partition. Follow these steps:
 
       #. ``uart=disabled``: this disables the serial port completely
       #. ``uart=bdf@<BDF value>``:  this sets the PCI serial port based on its BDF.
-         For example, use ``bdf@0:18.2`` for a BDF of 0:18.2 ttyS2.
+         For example, use ``bdf@0:18.1`` for a BDF of 0:18.1 ttyS1.
       #. ``uart=port@<port address>``: this sets the serial port address
+   #. ``vuart=ttySn@irqN``: this tells the hypervisor which virtual serial device SOS
+      will use and its IRQ number. This is used to avoid conflict with SOS passthrough  
+      devices' interrupt. If UART is set to ttyS1, and its native IRQ is 5, you'd better
+      set ``vuart=ttyS1@irq5`` (Use 'dmesg | grep tty' to get IRQ information).
```

Comment:
I'm having a bit of a hard time understanding this fully. Is it recommend, compulsory or purely optional to set the `vuart` parameter?

What are the rules for setting it, for example: does it need to match the UART used by the hypervisor console? Can you use any arbitrary `ttyS` number and IRQ number or does it have to be the parameters of a real UART device?

Reply:
I know that, it made the setting more complex.  Now for UEFI platform, they want to use one binary to support different boards. before just uart=xxx@address, and the vuart is default setting (by KConfig), but we found some boards could have problems for the IRQ sharing issues (not support in ACRN now), so we decided to add vuart setting to avoid that. Its setting is recommended to match with real HW uart and its IRQ to keep consistent.